src/test/ui/const-generics/min_const_generics/type-dependent/const-arg-in-const-arg.rs

```diff
@@ -0,0 +1,25 @@
+// run-pass
+#![feature(min_const_generics)]
+#![feature(const_fn)]
+
+struct Foo;
+
+impl Foo {
+    fn foo<const N: usize>(&self) -> usize {
+        let f = self;
+        f.bar::<{
+            let f = Foo;
+            f.bar::<7>()
+        }>() + N
+    }
+
+    const fn bar<const M: usize>(&self) -> usize {
```

Comment:
cc @rust-lang/wg-const-eval we're stabilizing generic consts in const fns together with `min_const_generics`. Unlike trait bounds on generic types, there's no question about what these values can do, they either just behave like runtime values or they are used in types, then their behaviour is bound by const eval anyway.